WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/details/ThreatDetailsViewModel.kt

```diff
@@ -0,0 +1,49 @@
+package org.wordpress.android.ui.jetpack.scan.details
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import org.wordpress.android.fluxc.model.scan.threat.ThreatModel
+import org.wordpress.android.ui.jetpack.common.JetpackListItemState
+import org.wordpress.android.ui.jetpack.scan.details.ThreatDetailsViewModel.UiState.Content
+import org.wordpress.android.ui.jetpack.usecases.GetThreatModelUseCase
+import javax.inject.Inject
+
+class ThreatDetailsViewModel @Inject constructor(
+    private val getThreatModelUseCase: GetThreatModelUseCase,
+    private val builder: ThreatDetailsListItemsBuilder
+) : ViewModel() {
+    private var isStarted = false
+    private var threatId: Long = 0
+
+    private val _uiState: MutableLiveData<UiState> = MutableLiveData()
+    val uiState: LiveData<UiState> = _uiState
+
+    fun start(threatId: Long) {
+        if (isStarted) {
+            return
+        }
+        isStarted = true
+        this.threatId = threatId
```

Comment:
Question (❓): Why is `threadId` exposed as a field, will it be used as such later on? Otherwise, I would suggest just passing the `threadId` parameter to the `getData()` function and remove the `threadId` field for now. Wdyt?

Reply:
`threatId` will be used later on to perform fix and ignore threat actions (API requires `threatId`) in the future.

Reply:
Great, thanks for the information Ashita! 🌟 

PS: I usually try to avoid exposing fields too soon, later on when it is going to be needed it would be a matter of a single refactor commit, so it doesn't seem as an overhead to me. As such, I usually do that when needed and avoid having the reviewer(s) asking me about any future intentions. Also, sometime this saves me from rolling back since what I expected happening in the future have now changed and I start being sad. 😝